chia/simulator/full_node_simulator.py

```diff
@@ -408,14 +409,19 @@ async def wait_transaction_records_entered_mempool(self, records: Collection[Tra
 
             await asyncio.sleep(0.050)
 
-    async def process_transaction_records(self, records: Collection[TransactionRecord]) -> None:
+    async def process_transactions(
+        self,
+        records: Collection[TransactionRecord] = (),
+        bundles: Collection[SpendBundle] = (),
+    ) -> None:
         """Process the specified transaction records and wait until they have been
         included in a block.
 
         Arguments:
             records: The transaction records to process.
+            bundles: The spend bundles to process.
         """
-        coins_to_wait_for: Set[Coin] = set()
+        coins_to_wait_for: Set[Coin] = {addition for bundle in bundles for addition in bundle.additions()}
```

Comment:
can you explain what this does ?
this syntax is confusing i would avoid it

Reply:
It reads better on multiple lines but yeah, there's a bit to be aware of.

```
x = {
    addition
    for bundle in bundles
    for addition in bundle.additions()
}
```
Is equivalent, at least in result, to this.
```
x = set()
for bundle in bundles:
    for addition in bundle.additions():
        x.add(addition)
```

Having the value (`addition`) first often trips people up in terms of ordering.  The order, or nesting, of the for loops is the same in both cases.  In this case this is being done to flatten all the 'additions' from the various spend bundles out into a single flat set.

Reply:
Or, Python's description.  https://docs.python.org/3.10/tutorial/datastructures.html#list-comprehensions

Reply:
ok yes the multiline is much clearer,  the first value was the thing that confused me.
personally i would avoid or maybe force the multiline for readability but it might just be me.

Reply:
I agree.  I used to say a lot that all my one liner's were four or five lines.  I don't think we can make black do that though.